src/Components/Facility/SpokeFacilityEditor.tsx

```diff
@@ -99,7 +100,7 @@ export default function SpokeFacilityEditor(props: SpokeFacilityEditorProps) {
             showNOptions={8}
             selected={selectedFacility}
             setSelected={(v) =>
-              v && !Array.isArray(v) && setSelectedFacility(v)
+              (v === null || (!Array.isArray(v) && v)) && setSelectedFacility(v)
```

Comment:
This isn't a big deal, but since you are specifying the type for state object to be FacilityModel or null, probably don't need the (&& v) check anymore.

Reply:
@NitinPSingh Like @Jacobjeevan mentioned, the logic wouldnt be effected but its hard to read, please simplify it to something like

```suggestion
              (v === null || !Array.isArray(v)) && setSelectedFacility(v)
```